lib/plugins/aws/deploy/index.js

```diff
@@ -55,6 +57,7 @@ class AwsDeploy {
               deploy: {
                 lifecycleEvents: [
                   'createStack',
+                  'ensureValidBucketExists',
```

Comment:
Do you think we need a new lifecycle event? Why not just do it at L130, before `setBucketName` ?

Reply:
I started with it in `checkForChanges`, but it didn't feel right to have logic that "modifies" something in step called `check`, which suggests that we're only doing checking here. But I see a point of not adding extra lifecycle events - what do you think is better here - having a little bit "inconsistent" (mutating) operation as a part of "check" logic or having a separate lifecycle event?

Reply:
Currently logic that resolves `bucketName` and crashes if there's no  bucket is in context of `checkForChanges`, so I think it's ok to add there recovery logic, which instead of crashing ensures the bucket

I'm also not sure about changing our current lifecycle events set, I'd rather treat it more as part of legacy design that's best if left untouched (and not further explored). To avoid unexpected issues I'd rather introduce new lifecycle events only if there's really hard to go without it. WDYT?

Reply:
That's a good point - let's treat lifecycle engine as "legacy" and not introduce new hooks unless there's a really good reason for it. I've changed the implementation.